extensions/Push/lib/Push.pm

```diff
@@ -50,15 +38,52 @@ sub start {
         $connector->backlog->reset_backoff();
     }
 
-    while(1) {
-        if ($self->_dbh_check()) {
-            $self->_reload();
-            $self->push();
+    my $pushd_loop = IO::Async::Loop->new;
+    my $main_timer = IO::Async::Timer::Periodic->new(
+        first_interval => 0,
+        interval       => POLL_INTERVAL_SECONDS,
+        reschedule     => 'drift',
+        on_tick        => sub {
+            if ( $self->_dbh_check() ) {
+                $self->_reload();
+                $self->push();
+            }
+        },
+    );
+    if ( Bugzilla->datadog ) {
+        my $dog_timer = IO::Async::Timer::Periodic->new(
+            interval   => 60 * 5,
+            reschedule => 'drift',
+            on_tick    => sub { $self->heartbeat },
+        );
+        $pushd_loop->add($dog_timer);
+        $dog_timer->start;
+    }
+
+    $pushd_loop->add($main_timer);
+    $main_timer->start;
+    $pushd_loop->run;
+}
+
+sub heartbeat {
+    my ($self) = @_;
+    my $dd = Bugzilla->datadog('pushd');
+
+    $dd->gauge('total', $self->_sql_count('SELECT COUNT(*) FROM push'));
+
+    foreach my $connector ($connectors->list) {
+        if ($connector->enabled) {
+            $dd->gauge("${name}_backlog", $self->_sql_count('SELECT COUNT(*) FROM push_backlog WHERE connector = ?', $name);
         }
-        sleep(POLL_INTERVAL_SECONDS);
     }
 }
 
+sub _sql_count {
+    my ($self, $sql, @bind) = @_;
+    my ($count) = Bugzilla->dbh->selectrow_array($sql, undef, @bind);
+    return $count;
```

Comment:
this seems unnecessary; there's nothing specific to this sub related to "counting", and `selectrow_array` called in a scalar context will return the desired result.

ie. replace
```
$dd->gauge('total', $self->_sql_count('SELECT COUNT(*) FROM push'));
```
with
```
$dd->gauge('total', Bugzilla->dbh->selectrow_array('SELECT COUNT(*) FROM push'));
```